db.go

```diff
@@ -2489,3 +2489,31 @@ func (d *DB) SetCreatorID(creatorID uint64) error {
 func (d *DB) ObjProvider() objstorage.Provider {
 	return d.objProvider
 }
+
+func (d *DB) checkVirtualBounds(m *fileMetadata, opts *IterOptions, iterOpts internalIterOpts) {
+	if !invariants.Enabled {
+		return
+	}
+
+	it, _, err := d.newIters(context.TODO(), m, opts, iterOpts)
+	if err != nil {
+		panic("pebble: error retrieving virtual iterator")
+	}
+
+	// check that virtual sstable bounds are tight from both ends.
+	first, _ := it.First()
+	if d.cmp(first.UserKey, m.Smallest.UserKey) != 0 {
+		panic("pebble: virtual sstable lower bound is not tight")
```

Comment:
would also make sense to print out the two keys in question here, as well as the file number

```
panic(fmt.Sprintf("pebble: virtual sstable %s lower bound is not tight: %s != %s", m.FileNum, m.Smallest.UserKey, first.UserKey))
```